packages/teams-js/test/internal/appId.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { AppId } from '../../src/internal/appId';
+import { maximumValidAppIdLength } from '../../src/internal/appIdValidation';
+
+// Since there are plenty of tests validating the individual validation functionality, these tests are intentionally not as
+// comprehensive as those. It executes a few basic tests and also validates that the error messages thrown are as expected.
+describe('doesStringContainNonPrintableCharacters', () => {
+  test('should not throw for "valid" app ids', () => {
+    expect(() => new AppId('8e6523aa-97f9-49ad-8614-75cae22f6597')).not.toThrow();
+    expect(() => new AppId('com.microsoft.teamspace.tab.youtube')).not.toThrow();
+  });
+
+  test('should throw error with "script" in message for app id containing script tag', () => {
+    expect(() => new AppId('<script>alert("Hello")</script>')).toThrowError(/script/);
```

Comment:
I don't think I've seen this syntax before or method of validating error messages (I think we usually match the entire error message string).

I like it; is this a recommended future approach and should we document it as guidance for how we want to unit test error messages?

Reply:
Sure, I can do that. I actually thing the right longer-term approach here must include a custom error object with an error code, not just a string. But that's a huge feature set all its own.